### ISSUE TITLE
PHP 8.2 support

### ISSUE THREAD

**Issue by jirismika** [prettier/plugin-php#2130]
**@prettier/plugin-php v0.19.2**
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCDoA6AEwgWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWsARhgOXBZoAEdBeGgA12V-PTkFEGIRCBhitlK2coMAXxYoCngYdBQ0eCxJUDZsGGaRNqrMMEFRvBAAHgB+YzAAHUx16DYKLFkkgAIY2WHsfYBZJNM5COQOACsoGJh94HX99-3pjg8Y-ZGIFT7AAkPku1ygAG51i11iAWkA)

**Input:**
```php
<?php

readonly class MyValueObject {
    public string $myValue;
}

```

**Output:**
```php
Invalid phpVersion value. Expected "5.0", "5.1", "5.2", "5.3", "5.4", "5.5", "5.6", "7.0", "7.1", "7.2", "7.3", "7.4", "8.0" or "8.1", but received "8.2".
```

**Expected:**
Not fail

Hi, 
I would like to ask what are the plans or roadmap to support PHP 8.2

**Comment by czosel** [comment]
Readonly class support landed in the parser recentily: https://github.com/glayzzle/php-parser/pull/1046
It should be relatively straight forward to make this work here (bumping the parser, minor changes in the printer). Contributions are welcome :wink:

**Comment by jirismika** [comment]
Well, the question was more general than only asking about readonly classes, but yes, readonly classes are something I am looking forward to use.\
Well, I tried to make it as you've challenged me, but from my attempts (and I haven't tried everything), it looks the parser is also missing support for [union and interception types](https://www.howtogeek.com/devops/whats-new-in-php-8.2/) ...

```php
private function foo((Countable&Iterator)|array $values) : void {
    // ...
}
```

```php
... SyntaxError: Parse Error : syntax error, unexpected '(', expecting T_VARIABLE on line 10
[error]    9 | 
[error] > 10 |   private function foo((Countable&Iterator)|array $values) : void {
```

Anyway, I gave it a try, but I failed terribly almost on each step as the contribution manual is not too detailed for me and prepared scripts doesn't work as expected (probably because expects global installations of some packages, while IMO it should use and install them locally) ... 
`yarn` - OK, `yarn test` - failed, because I dont have jest installed globally
Similar with `yarn prettier file.php` which wasn't found, but I was able to solve it with `./node_modules/.bin/prettier --plugin=. --parser=php file.php`

And about the printer, I am not able to debug it easily and I am lost in it.

I've invested some time into it, but most of the time was solving problems with the environment. The entry barrier is too high for me, sorry.

**Comment by genintho** [comment]
Readonly class support https://github.com/prettier/plugin-php/pull/2131 

(jest does not need to be installed globally, yarn should pick the folder version)

**Comment by czosel** [comment]
Readonly class support landed in `v0.19.3` :tada:

**Comment by Timbuktu1982** [comment]
Hi there,
i have installed the latest version of plugin-php (0.19.4), but i see the same message, if i use "8.2." for "phpVersion" in .prettierrc.js.
What can i do?
THX

**Comment by tomb1n0** [comment]
I've ran into the same issue as @Timbuktu1982 with updating the `phpVersion` in my `.prettierrc.json`

**Comment by czosel** [comment]
This should be fixed by #2179 - until it is released, you can just use "8.1" as phpVersion (which has the same behavior).

**Comment by czosel** [comment]
@Timbuktu1982 @tomb1n0 `v0.19.5` has just been released :tada:
